internal/graph/track_check_resolver_test.go

```diff
@@ -0,0 +1,280 @@
+			AuthorizationModelID: ulid.Make().String(),
+			TupleKey:             tuple.NewTupleKey("document:abc", "viewer", "user:somebody"),
+			RequestMetadata:      NewCheckRequestMetadata(20),
+		}
+		value, ok := trackChecker.loadModel(r)
+		require.NotNil(t, value)
+		require.False(t, ok)
+
+		path := "group:1#member@user"
+		sm := &sync.Map{}
+		sm.Store(
+			path,
+			&resolutionNode{
+				tm:   time.Now().Add(-trackerInterval),
+				hits: &atomic.Uint64{},
+			},
+		)
+
+		trackChecker.nodes.Store(trackerKey{store: ulid.Make().String(), model: ulid.Make().String()}, sm)
+		trackChecker.logExecutionPaths(false)
+
+		_, ok = sm.Load(path)
+		require.False(t, ok)
+	})
+
+	t.Run("tracker_success_launch_flush", func(t *testing.T) {
+		wg := sync.WaitGroup{}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(10) * time.Millisecond)
+		}()
+
+		path := "group:1#member@user"
+		sm := &sync.Map{}
+		sm.Store(
+			path,
+			&resolutionNode{
+				tm:   time.Now().Add(-trackerInterval),
+				hits: &atomic.Uint64{},
+			},
+		)
+
+		trackChecker.nodes.Store(trackerKey{store: ulid.Make().String(), model: ulid.Make().String()}, sm)
+
+		trackChecker.ticker.Reset(time.Duration(2) * time.Millisecond)
+		trackChecker.launchFlush()
+
+		wg.Wait()
+
+		_, ok := sm.Load(path)
+		require.False(t, ok)
+	})
+
+	t.Run("success_loadModel_and_loadPaths", func(t *testing.T) {
+		r := &ResolveCheckRequest{
+			StoreID:              ulid.Make().String(),
+			AuthorizationModelID: ulid.Make().String(),
+			TupleKey:             tuple.NewTupleKey("document:abc", "viewer", "user:somebody"),
+			RequestMetadata:      NewCheckRequestMetadata(20),
+		}
+		value, ok := trackChecker.loadModel(r)
+		require.NotNil(t, value)
+		require.False(t, ok)
+
+		path := trackChecker.getTK(tuple.NewTupleKey("folder:test", "reader", "user:somebody"))
+		trackChecker.loadPath(value, path)
+		trackChecker.addPathHits(r)
+
+		paths, ok := value.(*sync.Map)
+		require.True(t, ok)
+		_, ok = paths.Load(path)
+		require.True(t, ok)
+
+		paths, ok = value.(*sync.Map)
+		require.True(t, ok)
+		_, ok = paths.Load(path)
+		require.True(t, ok)
+	})
+
+	t.Run("success_limiter_disallow", func(t *testing.T) {
+		r := &ResolveCheckRequest{
+			StoreID:              ulid.Make().String(),
+			AuthorizationModelID: ulid.Make().String(),
+			TupleKey:             tuple.NewTupleKey("document:okta", "viewer", "user:alice"),
+			RequestMetadata:      NewCheckRequestMetadata(20),
+		}
+		value, ok := trackChecker.loadModel(r)
+		require.NotNil(t, value)
+		require.False(t, ok)
+
+		path := trackChecker.getTK(r.GetTupleKey())
+		paths, _ := value.(*sync.Map)
+
+		val, _ := paths.Load(path)
+
+		rn := val.(*resolutionNode)
+		rn.tm = time.Now().Add(-trackerInterval)
+		paths.Store(path, rn)
+
+		oldBurst := trackChecker.limiter.Burst()
+		trackChecker.limiter.SetBurst(0)
+
+		trackChecker.logExecutionPaths(false)
+		trackChecker.limiter.SetBurst(oldBurst)
```

Comment:
Nit: why bother setting the burst back to its old value? You can create a new trackChecker per test